libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1287,9 +1287,18 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		else
 		{
-			IRVariable argument = convert(*arguments.front(), *arrayType.baseType());
+			IRVariable argument =
+				arrayType.baseType()->isValueType() ?
+				convert(*arguments.front(), *arrayType.baseType()) :
+				*arguments.front();
+
+			string arrayPushFunction =
+				arrayType.baseType()->isValueType() ?
+				m_utils.storageArrayPushFunction(arrayType) :
+				m_utils.storageArrayPushFunction(arrayType, arguments.front()->annotation().type);
+
```

Comment:
```suggestion
			IRVariable argument = *arguments.front();
			if (arrayType.baseType()->isValueType())
				argument = convert(argument, arrayType.baseType());

			string arrayPushFunction =
				m_utils.storageArrayPushFunction(arrayType, argument.type());
```

Reply:
There is no copy constructor for IRVariable, so I needed to change this a bit now